mesonbuild/compilers/compilers.py

```diff
@@ -462,9 +467,50 @@ def __init__(self, compiler: T.Union['Compiler', StaticLinker],
                  iterable: T.Optional[T.Iterable[str]] = None):
         self.compiler = compiler
         self.__container = list(iterable) if iterable is not None else []  # type: T.List[str]
-        self.__seen_args = set()
-        for arg in self.__container:
-            self.__seen_args.add(arg)
+        self.pre = deque()
+        self.post = deque()
+
+    # Flush the saved pre and post list into the __container list
+    #
+    # This correctly deduplicates the entries after _can_dedup definition
+    # Note: This function is designed to work without delete operations, as deletions are worsening the performance a lot.
+    def flush_pre_post(self):
+        pre_flush = deque()
+        pre_flush_set = set()
+        post_flush = deque()
+        post_flush_set = set()
+
+        #The two lists are here walked from the front to the back, in order to not need removals for deduplication
+        self.pre.reverse()
+        for a in self.pre:
```

Comment:
You're not reusing the value of `self.pre` in sorted form, use `for a in sorted(self.pre)`, which is faster due to not doing an in-pace sort.

Reply:
Cool, thx.

Reply:
Okay wait, calling reserve() is faster here than calling sorted(self.pre, reverse=True) (we are talking about 0.01s in avg., so not much)

Reply:
oops. I really meant to say `reversed()` and said `sorted()`. The difference is obviously going to be bigger the larger the deque is, but generally in python you want to avoid calling `.sort()` or `.reverse()` on a container unless you need to iterate over that value multiple times.

Reply:
Okay yeah, that helped. Thank you :)